### PR TITLE
WT-7494 Add Python test to trigger update restore eviction during recovery

### PR DESCRIPTION
**Pull request by alisonfel**
Introduces a python suite test that enables the debug mode setting for update restore evict during recovery. The debug mode forces update restore eviction whenever we evict a page. We want to perform this in recovery to ensure that all the in-memory images have the proper write generation number and we don't end up reading stale transaction ID's stored on the page.

Let me know if I should expand the test in any way, wasn't super sure if I covered all the important behaviors/scenarios/edge cases :)

### CODE REVIEW

**Review by kommiharibabu — APPROVED** [review]
LGTM

File: test/suite/test_debug_mode10.py [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+        # the run write gen should be 1 at this point, equal to the connection-wide base write gen.
+        # Since we checkpointed after a series of large writes/updates, the write gen of the pages should
+        # definitely be greater than 1.
+        checkpoint_write_gen, checkpoint_run_write_gen = self.parse_write_gen("file:test_debug_mode10.wt")
+        self.assertEqual(checkpoint_run_write_gen, 1)
+        self.assertGreater(checkpoint_write_gen, checkpoint_run_write_gen)
+
+        olddir = "."
+        newdir = "RESTART"
+        self.simulate_crash(olddir, newdir)
```
[the remark this answers is not in the document]
Comment (alisonfel) [reply test/suite/test_debug_mode10.py]: Great point! I've updated the test to use `simulate_crash_restart` :)

File: test/suite/test_debug_mode10.py [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+            if not c.isdigit():
+                self.assertEqual(c, '=')
+                break
+            run_write_gen_str = c + run_write_gen_str
+
+        return int(write_gen_str), int(run_write_gen_str)
+
+    def test_update_restore_evict_recovery(self):
+        uri = 'table:test_debug_mode10'
+        nrows = 20000
```
[the remark this answers is not in the document]
Comment (alisonfel) [reply test/suite/test_debug_mode10.py]: That a fair point. I've reduced the number of rows by half `10000`, taking roughly ~3 seconds to complete (I hope thats ok?). I had to balance it out with the size of our updates in order to ensure we can reliably trigger `update restore eviction`

[the remark this answers is not in the document]
Comment (kommiharibabu) [reply test/suite/test_debug_mode10.py]: Thanks.

**Review by kommiharibabu — APPROVED** [review]
LGTM

File: test/suite/test_debug_mode10.py [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+from wtdataset import SimpleDataSet
+from helper import simulate_crash_restart
+
+# test_debug_mode10.py
+# Test the debug mode setting for update_restore_evict during recovery.
+# Force update restore eviction, whenever we evict a page. We want to
+# perform this in recovery to ensure that all the in-memory images have
+# the proper write generation number and we don't end up reading stale
+# transaction ID's stored on the page.
+class test_debug_mode10(wttest.WiredTigerTestCase):
```
Comment (quchenhao) [inline test/suite/test_debug_mode10.py:41]: I don't think this should be in debug mode test. I think it should be in recovery test.

**Review by quchenhao — APPROVED** [review]
just a small point. I think the test should be in test_rollback_to_stable as it is using the debug mode to test a specific scenario.

File: test/suite/test_debug_mode10.py (hunk 3 above)
Comment (alisonfel) [reply test/suite/test_debug_mode10.py]: Good point 👍 I've relabeled it as a rollback to stable test. 
Since it fits under the rts category I also updated the test to use the base rollback to stable class `test_rollback_to_stable_base`, in effect allowing us to remove our versions of the `large_updates` and `check` helpers.
